src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Policy/DefaultAuthorizationContext.cs

```diff
@@ -0,0 +1,68 @@
+﻿using CoreWCF.IdentityModel.Claims;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Text;
+
+namespace CoreWCF.IdentityModel.Policy
+{
+   internal class DefaultAuthorizationContext : AuthorizationContext
+    {
+        static DefaultAuthorizationContext empty;
+        SecurityUniqueId id;
+        ReadOnlyCollection<ClaimSet> claimSets;
+        DateTime expirationTime;
+        IDictionary<string, object> properties;
+
+        public DefaultAuthorizationContext(DefaultEvaluationContext evaluationContext)
+        {
+            this.claimSets = evaluationContext.ClaimSets;
+            this.expirationTime = evaluationContext.ExpirationTime;
+            this.properties = evaluationContext.Properties;
+        }
+
+        public static DefaultAuthorizationContext Empty
+        {
+            get
+            {
+                /* commenting 
+                if (LocalAppContextSwitches.EnableCachedEmptyDefaultAuthorizationContext)
+                {
```

Comment:
You can remove this commented code. This was put there as a back compat switch to revert the behavior of a fix. The problem was custom code was incorrectly modifying the empty default authotization context which resulted in leaking auth info from one call to another. The fix was to return a new one each time. My preferred fix at the time was to make it immutable and throw if trying to modify but there was concern that would be an even bigger breaking change. What are your thoughts? Should we make this an immutable singleton or allocate a new one each time?